src/HPolyhedron.jl

```diff
@@ -506,10 +558,20 @@ function remove_redundant_constraints!(P::PT;
         end
     end
 
-    deleteat!(P.constraints, setdiff(1:m, non_redundant_indices))
+    deleteat!(constraints, setdiff(1:m, non_redundant_indices))
     return true
 end
 
+function remove_redundant_constraints(constraints::AbstractVector{LinearConstraint{N}};
+                                      backend=GLPKSolverLP())::Bool where {N}
+    constraints_copy = copy(constraints)
+    if remove_redundant_constraints!(constraints_copy, backend=backend)
+        return constraints_copy
+    else # the constraints are infeasible
+        return N[]
```

Comment:
This does *not* represent an infeasible list of constraints! It is also not type-stable because the result should be a list of constraints.